frontend/src/hooks/useClients.ts

```diff
@@ -25,5 +25,12 @@ export function useClients() {
             .then(data => setClients(prevState => [...prevState, data]))
     }
 
-    return {clients, addClient, getAllClients}
+    function updateClient(client: Client) {
+        axios.put("/api/clients/" + client.id, client)
+            .then(response => response.data)
+            .then(data => setClients(prevState => [...prevState, data]))
```

Comment:
Attention: it seems to me like you get the old and also the updated client in your list this way! I think you will need to do a `prevState.map` where you check for the id. If it is the id of the changed client, return `data` instead of the old element (otherwise return the old element).